egg/platform_x86.py

```diff
@@ -836,6 +972,69 @@ def lnot1(simd_ext, typ):
             return 'return (__mmask{le})(~{in0});'.format(**fmtspec)
     return not1(simd_ext, typ, True)
 
+# -----------------------------------------------------------------------------
+## Code for constant true
+
+def true0(simd_ext, typ, logical=False):
+    if typ == 'f16':
+        return \
+        '''nsimd_{simd_ext}_v{logi}f16 ret;
+           nsimd_{simd_ext}_vf32 cte =
+             {pre}castsi{nbits}_ps({pre}set1_epi8(-1));
+           ret.v0 = cte;
+           ret.v1 = cte;
+           return ret;'''.format(logi='l' if logical else '', **fmtspec)
+    elif typ in ['f32', 'f64']:
+        return '''return {pre}castsi{nbits}{suf}(
+                           {pre}set1_epi8(-1));'''.format(**fmtspec)
+    else:
+        return '''return {pre}set1_epi8(-1);'''.format(**fmtspec)
+
+# -----------------------------------------------------------------------------
+## Code for constant logical true
+
+def ltrue0(simd_ext, typ):
+    if simd_ext in avx512:
+        if typ == 'f16':
+            return '''nsimd_{simd_ext}_vlf16 ret;
+                      ret.v0 = (__mmask16)~0;
+                      ret.v1 = (__mmask16)~0;
+                      return ret;'''.format(**fmtspec)
+        else:
+            return 'return (__mmask{le})~0;'.format(**fmtspec)
+    return true0(simd_ext, typ, True)
+
+# -----------------------------------------------------------------------------
+## Code for constant false
+
+def false0(simd_ext, typ, logical=False):
+    if typ == 'f16':
+        return \
+        '''nsimd_{simd_ext}_v{logi}f16 ret;
+           nsimd_{simd_ext}_vf32 cte = {pre}castsi{nbits}_ps({pre}set1_epi8(0));
+           ret.v0 = cte;
+           ret.v1 = cte;
+           return ret;'''.format(logi='l' if logical else '', **fmtspec)
+    elif typ in ['f32', 'f64']:
+        return '''return {pre}castsi{nbits}{suf}(
+                           {pre}set1_epi8(0));'''.format(**fmtspec)
+    else:
+        return '''return {pre}set1_epi8(0);'''.format(**fmtspec)
```

Comment:
For this function use the `*_setzero_*` family of intrinsics.

Reply:
Done.